### PR TITLE
[Docathon]: converted cudnn_rnn_determinism to markdown

### LINKED ISSUE #182462: [Docathon] Convert `cudnn_rnn_determinism.rst` from rST to MyST Markdown
## Description

Convert `docs/source/cudnn_rnn_determinism.rst` from reStructuredText to MyST Markdown and ensure that doc tests pass.

## Steps

1. **Assign the issue** to yourself using the `/assigntome` comment.

2. **Rename the file using `git mv`:**
   ```bash
   git mv docs/source/cudnn_rnn_determinism.rst docs/source/cudnn_rnn_determinism.md
   ```
   **Important!** Use `git mv` to convert the file — DO NOT delete the `.rst` and create a new `.md`.
   We won't accept PRs with deleted `.rst` files. We want to preserve the git log.

3. **Convert the content** from rST to MyST Markdown syntax.

   If you want to use AI for the conversion, we had very good results with
   [this converter script](https://github.com/svekars/sphinx-read-thedocs-test/blob/main/rst_to_myst_md_converter.py).

4. **Build the docs locally** as described in the [README](https://github.com/pytorch/pytorch/tree/main/docs).
   Preview locally with `make serve` to ensure the HTML renders correctly.

5. **Run lintrunner** from the root of the repository:
   ```bash
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

6. **Submit a PR** with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:
   - `pull / linux-docs / build-docs-python-false (pull_request)`
   - `linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)`

   Doc tests start after the Docker build, which might take about 30 minutes.
   The doc build itself should take about 25 minutes. When the build is done,
   you can see the preview of your build by clicking
   **📄 Preview Python docs built from this PR** in the pytorch-bot comment.

7. **Label your PR** with: `docathon-h1-2025`, `module: docs`, `topic: not user facing`, `topic: docs`.

8. **Compare the preview** of the pages you converted with the same pages on the main branch.
   The converted pages in preview must look the same as in the main branch.

9. Once the PR is accepted, one of the maintainers will merge it.

## References

- [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Quick-MyST-Conversion-Reference) (wiki)
- [MyST Parser documentation](https://myst-parser.readthedocs.io/)
- [Example PR](https://github.com/pytorch/pytorch/pull/155553)

cc @svekars @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by PyDevC**
Fixes: #182462

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

**Comment by PyDevC** [comment]
Renders correctly

2026-05-17-031627_hyprshot

**Comment by PyDevC** [comment]
@svekars I have update the include part but I can't test it locally, it does not raise any error and still works completely fine now that I have added the parser below `.. include:: cudnn_rnn_determinism.md` part.

This was the error around 2 commit.
https://github.com/pytorch/pytorch/actions/runs/25984781976/job/76461453712?pr=184080

**Comment by sekyondaMeta** [comment]
@claude

File: torch/nn/modules/rnn.py [hunk 1]
```diff
@@ -603,7 +603,8 @@ def forward(x, hx=None, batch_first=False):
     .. note::
         ``batch_first`` argument is ignored for unbatched inputs.
 
-    .. include:: ../cudnn_rnn_determinism.rst
+    .. include:: ../cudnn_rnn_determinism.md
```
Comment (svekars) [inline torch/nn/modules/rnn.py:606]: So, we need to check that it renders correctly on this page: https://docs-preview.pytorch.org/pytorch/pytorch/184080/generated/torch.nn.RNN.html (for some reason there is no preview) and compare to: https://docs.pytorch.org/docs/main/generated/torch.nn.RNN.html

Comment (PyDevC) [reply torch/nn/modules/rnn.py:606]: checked it and it does.
even tested after nuking whole doc/build directory.

**Review by svekars — APPROVED** [review]

**Comment by PyDevC** [comment]
@svekars @sekyondaMeta can you assign difficulty label to it? I would suggest a medium label since it required me to edit other files too
